spec/controllers/drawings_controller_spec.rb

```diff
@@ -1,5 +1,66 @@
 require 'rails_helper'
 
 RSpec.describe DrawingsController, type: :controller do
+  login_user
 
+  describe "POST create" do
+    let(:perform) { post :create, drawing: params }
+    let(:params) { FactoryGirl.attributes_for(:drawing) }
+
+    context "with valid params" do
+      it "creates new drawing" do
+        expect{ perform }.to change{ Drawing.count }.by(1)
+      end
+
+      it "redirects to index" do
+        expect(perform).to redirect_to drawings_path
+      end
+    end
+
+    context "with invalid params" do
+      let(:params) { { description: "This is not enough" } }
```

Comment:
The outer pair of braces encloses a single-line block for the `let` and the inner pair is the param hash. Does look confusing sometimes!